### DISCUSSION TITLE
Why for the first page only the `queryKey`'s `pagination` is a Proxy object and for the following pages not?

### DISCUSSION THREAD

**Discussion by frederikhors** [TanStack/query#7602] [Q&A]
_I opened https://github.com/sveltejs/svelte/issues/11854 thinking this is a Svelte 5 issue, but maybe is a tanstack one._

Steps to reproduce:

1. Open [the reproduction](https://stackblitz.com/edit/sveltejs-kit-template-default-mmkgy4?file=src%2Froutes%2F%2Blayout.svelte,src%2Froutes%2F%2Bpage.svelte,src%2Froutes%2FPagination.svelte,src%2Flib%2Ftypes.ts) 

1. Open the browser console

1. When the project starts it will errors with:

    ```
    Uncaught (in promise) DOMException: Failed to execute 'put' on 'IDBObjectStore': #<Object> could not be cloned.
    ```

    ![image](https://github.com/sveltejs/svelte/assets/41120635/042075ba-195a-4068-9ea3-c99d5e416532)

    This is happening because IndexedDB cannot serialize and persist a Proxy object which is `pagination` in this case.

    **But...**

1. If you click on "Next" button the pagination changes from `{"page":1,"size":5}` to `{"page":2,"size":5}` and IndexedDB persists the object with no issues!

1. Even stranger: if you click on the "Previous" button it changes from `page:2` to `page:1` and the error is here again!

![image](https://github.com/sveltejs/svelte/assets/41120635/87397aa2-93e7-471b-b6bd-4874303f075b)

Why for the first page only the `queryKey`'s `pagination` is a Proxy object and for the following pages not?

### How often does this bug happen?

Every time

### Tanstack Query adapter

svelte-query

### TanStack Query version

5

### TypeScript version

5

**Comment by dummdidumm** [comment]
This has nothing to do with svelte-query

**Comment by frederikhors** [comment]
> This has nothing to do with svelte-query

Oh, hey! Thanks for your answer. Do you think it's a Svelte 5 problem?
